app/models/charge.rb

```diff
@@ -4,6 +4,7 @@ class Charge < ActiveRecord::Base
   delegate :balance, to: :user
 
   after_save :update_balance_amount
+  after_save :update_money_charged
```

Comment:
after save there is `update_money_charged` should be called, but you didn't define it.
Define this method and add logic, that updates field `money_charged` for charge's balance